src/main/java/ch/njol/skript/expressions/ExprStringCase.java

```diff
@@ -41,76 +44,118 @@
+				"%strings% in [(0¦lenient|1¦strict) ](proper|title)[ ]case",
+				"[(0¦lenient|1¦strict) ](proper|title)[ ]case %strings%",
+				"%strings% in [(0¦lenient|1¦strict) ]camel[ ]case",
+				"[(0¦lenient|1¦strict) ]camel[ ]case %strings%",
+				"%strings% in [(0¦lenient|1¦strict) ]pascal[ ]case",
+				"[(0¦lenient|1¦strict) ]pascal[ ]case %strings%",
+				"%strings% in [(1¦lower|2¦upper|3¦capital|4¦screaming)[ ]]snake[ ]case",
+				"[(1¦lower|2¦upper|3¦capital|4¦screaming)[ ]]snake[ ]case %strings%",
+				"%strings% in [(1¦lower|2¦upper|3¦capital)[ ]]kebab[ ]case",
+				"[(1¦lower|2¦upper|3¦capital)[ ]]kebab[ ]case %strings%");
 	}
 	
-	@Nullable
-	private Expression<String> origin;
-	@Nullable
-	private String literal;
-	private int mode;
+	@SuppressWarnings("null")
+	private Expression<String> expr;
+	
+	/* 0: No Change, 1: Upper Case, 2: Lower Case, 3: Strict */
+	private int casemode = 0; // Defaults to No Change 
 	
-	@SuppressWarnings({"unchecked"})
+	/* 0: Basic Case Change, 1: Proper Case, 2: Camel Case, 3: Pascal Case, 4: Snake Case, 5: Kebab Case */
+	private int type = 0; // Defaults to Basic Case Change 
+	
+	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		if (exprs[0] instanceof Literal)
-			literal = ((Literal<String>) exprs[0]).getSingle();
-		else
-			origin = (Expression<String>) exprs[0];
-		if (matchedPattern == 1)
-			mode = UPPER;
-		else
-			mode = parseResult.mark;
-		
+		expr = (Expression<String>) exprs[0];
+		if (matchedPattern <= 1) { // Basic Case Change 
+			casemode = (parseResult.mark == 0) ? 1 : 2;
+		} else if (matchedPattern == 2) { // Basic Case Change 
+			casemode = 2;
+		} else if (matchedPattern <= 4) { // Proper Case 
+			type = 1;
+			if (parseResult.mark != 0)
+				casemode = 3;
+		} else if (matchedPattern <= 6) { // Camel Case 
+			type = 2;
+			if (parseResult.mark != 0)
+				casemode = 3;
+		} else if (matchedPattern <= 8) { // Pascal Case 
+			type = 3;
+			if (parseResult.mark != 0)
+				casemode = 3;
+		} else if (matchedPattern <= 10) { // Snake Case 
+			type = 4;
+			if (parseResult.mark != 0)
+				casemode = (parseResult.mark == 1) ? 2 : 1;
+		} else if (matchedPattern <= 12) { // Kebab Case 
+			type = 5;
+			if (parseResult.mark != 0)
+				casemode = (parseResult.mark == 1) ? 2 : 1;
+		}
 		return true;
 	}
 	
+	@SuppressWarnings("null")
 	@Override
 	@Nullable
 	protected String[] get(Event e) {
-		String str;
-		if (literal != null)
-			str = changeCase(literal, mode);
-		else if (origin != null)
-			str = changeCase(origin.getSingle(e), mode);
-		else
-			str = "";
-		
-		return new String[] {str};
+		String[] strs = expr.getAll(e);
+		for (int i = 0; i < strs.length; i++) {
+			if (strs[i] != null) {
+				switch (type) {
+					case 0: // Basic Case Change 
+						strs[i] = (casemode == 1) ? strs[i].toUpperCase(Locale.ENGLISH) : strs[i].toLowerCase(Locale.ENGLISH);
```

Comment:
`Locale.ENGLISH` shouldn't be used in such case, because the string is not going to be used for something that requires English like an enum name. For example you would except "isim" to return `İSİM`, not `ISIM` for Turkish.